### PR TITLE
fix(parental-leave):  Change type when applying for Dvalarstyrk

### PR DESCRIPTION
**Pull request by birkirkristmunds**
[ https://dit-iceland.atlassian.net/browse/UUFV-435]( https://dit-iceland.atlassian.net/browse/UUFV-435)

## Why

Fix the type sent to VMST when applying for Dvalarstyrkur

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **Refactor**
	- Streamlined the application process in the Parental Leave service, enhancing the flow and simplifying state management.
- **New Features**
	- Enhanced state-specific logic handling in the Parental Leave module to better accommodate changes related to employer and period adjustments.
- **Improvements**
	- Updated the Parental Leave Template to improve application validation and submission processes.

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts [hunk 1]
```diff
@@ -1415,7 +1408,7 @@ export class ParentalLeaveService extends BaseTemplateApiService {
         periods,
         attachments,
         false, // put false in testData as this is not dummy request
-        checkActionName(application, actionNameFromParams),
+        checkActionName(application, params),
```
Comment (karenbjorg) [inline libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts:1411]: params is always undefined, cannot it be removed?

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts [hunk 2]
```diff
@@ -492,23 +493,14 @@ export const isDateInTheFuture = (date: string) => {
...
   application: ApplicationWithAttachments,
-  params: FileType | undefined = undefined,
+  params: any | undefined = undefined,
 ) => {
-  const { actionName } = getApplicationAnswers(application.answers)
+  const { state } = application
+  const { addEmployer, addPeriods, changeEmployer, changePeriods, actionName } =
+    getApplicationAnswers(application.answers)
+
   if (params) {
```
Comment (karenbjorg) [inline libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts:504]: remove this check? params is never set.

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts [hunk 3]
```diff
@@ -517,6 +509,29 @@ export const checkActionName = (
       params === 'employer'
     return params
   }
+
```
Comment (karenbjorg) [inline libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts:512]: do we need to setActionName in the template and just use the findActionName logic in the checkActionName function?

Comment (karenbjorg) [reply libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts:512]: and if so, rename the checkActionName function to something more descriptive..

Comment (birkirkristmunds) [reply libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts:508]: We need to use setActionName to keep bookkeeping of what actionName has been selected when sending application

File: libs/application/templates/parental-leave/src/lib/parentalLeaveUtils.ts [hunk 4]
```diff
@@ -2109,3 +2111,71 @@ export const getSelectOptionLabel = (options: SelectOption[], id?: string) => {
...
+    ) {
+      return FileType.EMPPER
+    } else if (changeEmployer || addEmployer === YES) {
+      return FileType.EMPLOYER
+    } else if (changePeriods || addPeriods === YES) {
+      return FileType.PERIOD
+    }
+  }
+  return undefined
+}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/parental-leave/src/lib/parentalLeaveUtils.ts]: Fxied

File: libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts (hunk 2 above)
Comment (birkirkristmunds) [reply libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.utils.ts]: Done

**Pull request merged**